app/src/server/api/routers/marriage.ts

```diff
@@ -0,0 +1,221 @@
+    .output(baseServerResponse)
+    .mutation(async ({ ctx, input }) => {
+      const request = await fetchRequest(ctx.drizzle, input.id, "MARRIAGE");
+      if (request.senderId !== ctx.userId) {
+        return errorResponse("You can only cancel requests created by you");
+      }
+      if (request.status !== "PENDING") {
+        return errorResponse("You can only cancel pending requests");
+      }
+      void pusher.trigger(request.receiverId, "event", { type: "MARRIAGE" });
+      return await updateRequestState(ctx.drizzle, input.id, "CANCELLED", "MARRIAGE");
+    }),
+  acceptRequest: protectedProcedure
+    .input(z.object({ id: z.string() }))
+    .output(baseServerResponse)
+    .mutation(async ({ ctx, input }) => {
+      // Fetch
+      const request = await fetchRequest(ctx.drizzle, input.id, "MARRIAGE");
+      // Secondary fetches
+      const [sender, receiver, senderAssociations, receiverAssociations] =
+        await Promise.all([
+          fetchUser(ctx.drizzle, request.senderId),
+          fetchUser(ctx.drizzle, request.receiverId),
+          fetchAssociations(ctx.drizzle, request.senderId, "MARRIAGE"),
+          fetchAssociations(ctx.drizzle, request.receiverId, "MARRIAGE"),
+        ]);
+      // Guards
+      if (ctx.userId !== request.receiverId) return errorResponse("Not your request");
+      if (sender.marriageSlots <= senderAssociations.length)
+        return errorResponse("Sender does not have enough marriage slots");
+      if (receiver.marriageSlots <= receiverAssociations.length)
+        return errorResponse("Receiver does not have enough marriage slots");
+      if (
+        senderAssociations.length > 0 &&
+        senderAssociations.filter(
+          (x) => x.userOne.userId == ctx.userId || x.userTwo.userId == ctx.userId,
+        )
+      )
+        return errorResponse("You are already married to this user");
+      // Mutate
+      await Promise.all([
+        updateRequestState(ctx.drizzle, input.id, "ACCEPTED", "MARRIAGE"),
+        insertAssociation(
+          ctx.drizzle,
+          request.senderId,
+          request.receiverId,
+          "MARRIAGE",
+        ),
+      ]);
+      void pusher.trigger(request.senderId, "event", { type: "MARRIAGE" });
+      // Create
+      return { success: true, message: "Proposal Accepted" };
+    }),
+  getMarriedUsers: protectedProcedure
+    .input(z.object({ id: z.string().optional() }))
+    .query(async ({ ctx, input }) => {
+      const associations = await fetchAssociations(
+        ctx.drizzle,
+        input.id ?? ctx.userId,
+        "MARRIAGE",
+      );
+      const marriedUsers = associations.map((x) =>
+        x.userOne.userId !== input.id ?? ctx.userId ? x.userOne : x.userTwo,
+      );
+
+      return marriedUsers;
+    }),
+  getDivorcedAssociations: protectedProcedure.query(async ({ ctx }) => {
+    return await fetchAssociations(ctx.drizzle, ctx.userId, "DIVORCED");
+  }),
+  updateAssociation: protectedProcedure
```

Comment:
Let's call this endpoint divorce for clarity?

Reply:
Renamed, originally I figured this would get re-used if we utilized associations for non-marriage things but realistically we could refactor it into an associations api. Same thing for requests, I think most of that lives in sparing atm.